testutil/ciao-down/ciao_down.go

```diff
@@ -0,0 +1,320 @@
+		fmt.Printf("Downloaded %d MB of %d\n", p.downloadedMB, p.totalMB)
+	} else {
+		fmt.Printf("Downloaded %d MB\n", p.downloadedMB)
+	}
+}
+
+func prepare(ctx context.Context, errCh chan error) {
+	fmt.Println("Checking environment")
+	memGB, CPUs, debug, err := prepareFlags()
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	ws, err := prepareEnv(ctx)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	_, err = os.Stat(ws.instanceDir)
+	if err == nil {
+		errCh <- fmt.Errorf("instance already exists")
+		return
+	}
+
+	fmt.Println("Installing host dependencies")
+	installDeps(ctx)
+
+	err = os.MkdirAll(ws.instanceDir, 0755)
+	if err != nil {
+		errCh <- fmt.Errorf("unable to create cache dir: %v", err)
+		return
+	}
+
+	failed := true
+	defer func() {
+		if failed {
+			_ = os.RemoveAll(ws.instanceDir)
+		}
+	}()
+
+	qcowPath, err := downloadUbuntu(ctx, ws.ciaoDir, downloadProgress)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	err = buildISOImage(ctx, ws.instanceDir, ws, debug)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	err = createRootfs(ctx, qcowPath, ws.instanceDir)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	fmt.Printf("Booting VM with %d GB RAM and %d cpus\n", memGB, CPUs)
+
+	err = bootVM(ctx, ws, memGB, CPUs)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	err = manageInstallation(ctx, ws.instanceDir, ws)
+	if err != nil {
+		errCh <- err
+		return
+	}
+	errCh <- nil
+	failed = false
+	fmt.Println("VM successfully created!")
+	fmt.Println("Type ciao-down connect to start using it.")
+}
+
+func start(ctx context.Context, errCh chan error) {
+	memGB, CPUs, err := startFlags()
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	ws, err := prepareEnv(ctx)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	fmt.Printf("Booting VM with %d GB RAM and %d cpus\n", memGB, CPUs)
+
+	err = bootVM(ctx, ws, memGB, CPUs)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	if err == nil {
+		fmt.Println("VM Started")
+	}
+
+	errCh <- err
+}
+
+func stop(ctx context.Context, errCh chan error) {
+	ws, err := prepareEnv(ctx)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	err = stopVM(ctx, ws.instanceDir)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	if err == nil {
+		fmt.Println("VM Stopped")
+	}
+
+	errCh <- err
+}
+
+func quit(ctx context.Context, errCh chan error) {
+	ws, err := prepareEnv(ctx)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	err = quitVM(ctx, ws.instanceDir)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	if err == nil {
+		fmt.Println("VM Quit")
+	}
+
+	errCh <- err
+}
+
+func status(ctx context.Context, errCh chan error) {
+	ws, err := prepareEnv(ctx)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	statusVM(ctx, ws.instanceDir)
+	errCh <- err
+}
+
+func connect(errCh chan error) {
+	path, err := exec.LookPath("ssh")
+	if err != nil {
+		errCh <- fmt.Errorf("Unable to locate ssh binary")
+	}
+
+	err = syscall.Exec(path, []string{path, "127.0.0.1", "-p", "10022"},
```

Comment:
@markdryan does it make sense to ensure the key does not get added to the known hosts
ssh -T -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i